src/pages/RFOX/components/Stake/Bridge/hooks/useRfoxBridge.ts

```diff
@@ -46,6 +52,17 @@ type UseRfoxBridge = (props: UseRfoxBridgeProps) => {
   serializedL2TxIndex: string | undefined
 }
 
+const isTradeQuoteQueryEnabled = (
+  input: ArbitrumBridgeTradeQuoteInput & { wallet: HDWallet | null },
+): input is GetEvmTradeQuoteInputWithWallet =>
+  Boolean(
+    input.sellAsset &&
+      input.buyAsset &&
+      input.chainId &&
+      input.accountNumber !== undefined &&
+      input.wallet,
+  )
+
```

Comment:
The third and final guts of this - this overcomes TS type system being derp, `isTradeQuoteQueryEnabled` being a boolean won't cut it